app/grandchallenge/hanging_protocols/forms.py

```diff
@@ -116,3 +129,86 @@ def _validate_slice_plane_indicator(self, *, viewport, viewport_names):
+                self.fields[
+                    "view_content"
+                ].help_text += f"The following interfaces are used in your {self.instance._meta.verbose_name}: {oxford_comma(interface_slugs)}. "
+
+            view_content_example = self.generate_view_content_example()
+
+            if view_content_example:
+                self.fields[
+                    "view_content"
+                ].help_text += f"Example usage: {view_content_example}. "
+            else:
+                self.fields[
+                    "view_content"
+                ].help_text += "No interfaces of type (image, chart, pdf, mp4, thumbnail_jpg, thumbnail_png) are used. At least one interface of those types is needed to config the viewer. "
+
+        self.fields["view_content"].help_text += format_lazy(
+            'Refer to the <a href="{}">documentation</a> for more information',
+            reverse("documentation:detail", args=["viewer-content"]),
+        )
+
+    def generate_view_content_example(self):
+        images = list(
+            self.instance.interfaces.filter(
+                kind=InterfaceKindChoices.IMAGE
+            ).values_list("slug", flat=True)
+        )
+        mandatory_isolation_interfaces = list(
+            self.instance.interfaces.filter(
+                kind__in=InterfaceKind.interface_type_mandatory_isolation()
+            ).values_list("slug", flat=True)
+        )
+
+        if not images and not mandatory_isolation_interfaces:
+            return None
+
+        overlays = list(
+            self.instance.interfaces.exclude(
+                kind__in=InterfaceKind.interface_type_undisplayable()
+                | InterfaceKind.interface_type_mandatory_isolation()
+                | {InterfaceKindChoices.IMAGE}
+            ).values_list("slug", flat=True)
+        )
+        if images:
+            overlays_per_image = len(overlays) // len(images)
+            remaining_overlays = len(overlays) % len(images)
+
+        view_content_example = {}
+
+        for port in ViewportNames.values:
+            if mandatory_isolation_interfaces:
+                view_content_example[port] = [
+                    mandatory_isolation_interfaces.pop(0)
+                ]
+            elif images:
+                view_content_example[port] = [images.pop(0)]
+                for _ in range(overlays_per_image):
+                    view_content_example[port].append(overlays.pop(0))
+                if remaining_overlays > 0:
+                    view_content_example[port].append(overlays.pop(0))
+                    remaining_overlays -= 1
+
+        try:
+            JSONValidator(schema=VIEW_CONTENT_SCHEMA)(
+                value=view_content_example
+            )
+            self.instance.clean_view_content(view_content_example)
+        except ValidationError as e:
+            raise RuntimeError(
+                f"view_content example is not valid. Validation error: {e.messages}"
+            )

```

Comment:
The way to do this so that context is not lost is

```python
try:
    ...
except InterestingError as error:
    raise NewError("New message") from error
```

See https://docs.astral.sh/ruff/rules/raise-without-from-inside-except/

Reply:
Thanks, James!

Reply:
Thanks for the feedback!
Points taken.
Done.